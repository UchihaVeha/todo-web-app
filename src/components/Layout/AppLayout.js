// @flow
import * as React from 'react';
import { compose, withState, withHandlers, pure } from 'recompose';
import { withStyles } from 'material-ui/styles';
import { IconButton, AppBar, Toolbar } from 'material-ui';
import { grey } from 'material-ui/colors';
import MenuIcon from 'material-ui-icons/Menu';
import AppMenu from 'components/AppMenu';

const upMdWith = 960;

const styles = theme => ({
  root: {
    display: 'flex',
    minHeight: '100vh',
    flexDirection: 'column'
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    flexGrow: 1,
    background: grey['100']
  },
  header: {
    [theme.breakpoints.up('md')]: {
      maxWidth: upMdWith,
      margin: '0 auto'
    }
  },
  toolbar: {
    alignItems: 'stretch',
    flexDirection: 'column'
  },
  content: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    flexGrow: 1,
    [theme.breakpoints.up('md')]: {
      maxWidth: upMdWith,
      margin: '0 auto'
    }
  }
});

type Props = {
  appBar: React.ComponentType<*>,
  content: React.ComponentType<*>,
  classes: Return<typeof styles>,
  handleMenuToggle: () => void,
  open: boolean
};
export const AppLayoutComponent = ({
  appBar: Header,
  content: Content,
  classes,
  handleMenuToggle,
  open
}: Props) => (
  <div className={classes.root}>
    <div className={classes.container}>
      <AppMenu open={open} handleMenuToggle={handleMenuToggle} />
      <AppBar position="static" className={classes.header}>
        <Toolbar>
          <IconButton
            color="contrast"
            aria-label="Menu"
            onClick={handleMenuToggle}
          >
            <MenuIcon />
          </IconButton>
        </Toolbar>
        <Toolbar className={classes.toolbar}>
          <Header />
        </Toolbar>
      </AppBar>
      <div className={classes.content}>
        <Content />
      </div>
    </div>
  </div>
);

export default compose(
  pure,
  withState('open', 'setOpen', false),
  withHandlers({
    handleMenuToggle: ({ open, setOpen }) => () => setOpen(!open)
  }),
  withStyles(styles, { name: 'AppLayout' })
)(AppLayoutComponent);
