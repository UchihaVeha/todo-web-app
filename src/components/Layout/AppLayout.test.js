import React from 'react';
import { shallow } from 'enzyme';
import AppLayout, { AppLayoutComponent } from './AppLayout';

describe('Layout/AppLayout component', () => {
  const wrapper = (
    {
      open = false,
      handleMenuToggle = jest.fn(),
      appBar = () => <div>appBar</div>,
      content = () => <div>content</div>,
      classes = {}
    } = {}
  ) =>
    shallow(
      <AppLayoutComponent
        appBar={appBar}
        content={content}
        open={open}
        handleMenuToggle={handleMenuToggle}
        classes={classes}
      />
    );

  it('should render component', () => {
    expect(wrapper()).toMatchSnapshot();
  });

  it('should render with enhances', () => {
    expect(
      shallow(
        <AppLayout
          appBar={() => <div>appBar</div>}
          content={() => <div>content</div>}
        />
      )
    ).toMatchSnapshot();
  });
});
