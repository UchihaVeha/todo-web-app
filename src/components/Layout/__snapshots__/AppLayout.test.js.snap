// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/AppLayout component should render component 1`] = `
<div>
  <div>
    <pure(withStyles(AppMenuComponent))
      handleMenuToggle={[Function]}
      open={false}
    />
    <withStyles(AppBar)
      position="static"
    >
      <withStyles(Toolbar)>
        <withStyles(IconButton)
          aria-label="Menu"
          color="contrast"
          onClick={[Function]}
        >
          <pure(Menu) />
        </withStyles(IconButton)>
      </withStyles(Toolbar)>
      <withStyles(Toolbar)>
        <Component />
      </withStyles(Toolbar)>
    </withStyles(AppBar)>
    <div>
      <Component />
    </div>
  </div>
</div>
`;

exports[`Layout/AppLayout component should render with enhances 1`] = `
<withState(withHandlers(withStyles(AppLayoutComponent)))
  appBar={[Function]}
  content={[Function]}
/>
`;
